### PR TITLE
HTTP mirroring: add an option to follow HTTP redirections

### PR DESCRIPTION
**Pull request by dclaisse**

### CODE REVIEW

File: mirror/modules/sink/http.go [hunk 1]
```diff
@@ -42,9 +42,10 @@ func init() {
 }
 
 type HTTPConfig struct {
-	TargetURL *expr.StringExpr `json:"target_url,omitempty"`
-	Timeout   string           `json:"timeout,omitempty"`
-	Parallel  int              `json:"parallel"`
+	FollowRedirect string           `json:"follow_redirect,omitempty"`
```
Comment (ShimmerGlass) [inline mirror/modules/sink/http.go:45]: Can be a bool. This way you won't have to parse the string, and you can use true/false in your json

File: mirror/modules/sink/http.go [hunk 2]
```diff
@@ -67,6 +68,18 @@ func NewHTTP(ctx *mirror.ModuleContext, cfg []byte) (mirror.Module, error) {
 		return nil, err
 	}
 
+	followRedirect, err := strconv.ParseBool(c.FollowRedirect)
+	if err != nil {
+		return nil, fmt.Errorf("follow_redirect: %w", err)
+	}
+
+	checkRedirect := (func(req *http.Request, via []*http.Request) error)(nil)
```
Comment (ShimmerGlass) [inline mirror/modules/sink/http.go:76]: Would be cleaner to 

1. create the http client (move from HTTP struct creation)
2. if !followRedirect then set the call back
3. use http client var in HTTP struct creation

Btw your condition should be reversed, set the callback if followRedirect is false

**Review by ShimmerGlass — CHANGES_REQUESTED** [review]

File: mirror/modules/sink/http.go (hunk 1 above)
Comment (dclaisse) [reply mirror/modules/sink/http.go]: Indeed, much nicer.

File: mirror/modules/sink/http.go (hunk 2 above)
Comment (dclaisse) [reply mirror/modules/sink/http.go]: Thanks for the advice, it's indeed much more readable.

File: README.md [hunk 3]
```diff
@@ -128,6 +128,7 @@ or
 {
   "type": "sink.http",
   "config": {
+    "follow_redirect": "true",
```
Comment (ShimmerGlass) [inline README.md:131]:
```suggestion
    "follow_redirect": true,
```

File: mirror/modules/sink/http.go [hunk 4]
```diff
@@ -77,17 +78,22 @@ func NewHTTP(ctx *mirror.ModuleContext, cfg []byte) (mirror.Module, error) {
 		maxWorkers = c.Parallel
 	}
 
+	http_client := &http.Client{
```
Comment (ShimmerGlass) [inline mirror/modules/sink/http.go:81]:
```suggestion
	httpClient := &http.Client{
```

File: mirror/modules/sink/http.go [hunk 5]
```diff
@@ -77,17 +78,22 @@ func NewHTTP(ctx *mirror.ModuleContext, cfg []byte) (mirror.Module, error) {
...
+		Timeout: timeout,
+	}
+
+	if c.FollowRedirect == false {
```
Comment (ShimmerGlass) [inline mirror/modules/sink/http.go:85]:
```suggestion
	if !c.FollowRedirect {
```

**Review by ShimmerGlass — CHANGES_REQUESTED** [review]

**Comment by ShimmerGlass** [comment]
Bit of a nitpick but the option should be called `follow_redirects` IMHO, since they are many of them

File: mirror/modules/sink/http.go (hunk 4 above)
Comment (dclaisse) [reply mirror/modules/sink/http.go]: Agreed

File: mirror/modules/sink/http.go (hunk 5 above)
Comment (dclaisse) [reply mirror/modules/sink/http.go]: Agreed

**Comment by dclaisse** [comment]
> Bit of a nitpick but the option should be called `follow_redirects` IMHO, since they are many of them

Indeed I started naming it in a plural form then got confused by "CheckRedirect" method name, but all in all it makes sense, CheckRedirect is for one redirect, so the option should indeed be named follow_redirects as there could be many.

File: README.md (hunk 3 above)
Comment (dclaisse) [reply README.md]: Agreed

**Review by ShimmerGlass — APPROVED** [review]
LGTM, thx

**Pull request merged**